util/src/main/java/com/farao_community/farao/util/MultipleNetworkPool.java

```diff
@@ -21,16 +20,49 @@
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
 public class MultipleNetworkPool extends AbstractNetworkPool {
+
+    private int networkNumberOfClones;
+
     protected MultipleNetworkPool(Network network, String targetVariant, int parallelism) {
         super(network, targetVariant, parallelism);
+        this.networkNumberOfClones = 0;
+    }
+
+    @Override
+    protected void cleanVariants(Network networkClone) {
+        List<String> variantsToBeRemoved = networkClone.getVariantManager().getVariantIds().stream()
+                .filter(variantId -> !variantId.equals(VariantManagerConstants.INITIAL_VARIANT_ID))
+                .filter(variantId -> !variantId.equals(stateSaveVariant))
+                .collect(Collectors.toList());
+        variantsToBeRemoved.forEach(variantId -> networkClone.getVariantManager().removeVariant(variantId));
+    }
+
+    @Override
+    public int getNetworkNumberOfClones() {
+        // The number of clones includes the original network itself
+        return networkNumberOfClones;
     }
 
     @Override
-    protected void initAvailableNetworks(Network network) {
-        TECHNICAL_LOGS.debug("Filling network pool with copies of network '{}' on variant '{}'", network.getId(), targetVariant);
+    public void addNetworkClones(int numberOfClones) {
```

Comment:
rename this to initClones(int desiredNumberOfClones)
inside method, add addedClones = min(parallelism, desiredNumberOfClones) - networkNumberOfClones